File: src/1441-Build-an-Array-With-Stack-Operations/1441.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string> res;
        int k = 1;

        for (int i = 0; i < target.size(); i++, k++) {
            if (target[i] == k) { 
                res.emplace_back("Push");
            } else { 
                i--;
                res.emplace_back("Push");
                res.emplace_back("Pop");
            }
        }
        return res;
    }
};

File: src/1441-Build-an-Array-With-Stack-Operations/1441.js
var buildArray = function(target, n) {
    let res = [], k = 1;

    for (let i = 0; i < target.length; i++, k++) {
        if (target[i] == k) { 
            res.push("Push");
        } else { 
            i--;
            res.push("Push");
            res.push("Pop");
        }
    }
    return res;
};